### PR TITLE
Add route assets to preload

### PR DESCRIPTION
**Pull request by oleh-momot**

### CODE REVIEW

**Comment by oleh-momot** [comment]
It depends on https://github.com/StyleT/tailorx/pull/19

File: ilc/common/router/ServerRouter.js [hunk 1]
```diff
@@ -43,6 +41,40 @@ module.exports = class ServerRouter {
         }
     }
 
+    async getRouteAssets(reqUrl) {
```
Comment (StyleT) [inline ilc/common/router/ServerRouter.js]: I think it's better to have this code in a form of private method inside `configs-injector.js`
It has little in common with routing

File: ilc/common/router/ServerRouter.js [hunk 2]
```diff
@@ -43,6 +41,40 @@ module.exports = class ServerRouter {
...
+            const appInfo = this.__apps[slotData.appName];
+
+            if (!routeAssets.includes(appInfo.spaBundle)) {
+                routeAssets.push(appInfo.spaBundle);
+            }
+
+            Object.keys(appInfo.dependencies).filter((dependencyName) => {
+                const dependency = appInfo.dependencies[dependencyName];
+
+                if (!(/\.js$/.test(dependency))) {
```
Comment (StyleT) [inline ilc/common/router/ServerRouter.js]: str2.endsWith('?')

File: ilc/common/router/ServerRouter.js [hunk 3]
```diff
@@ -43,6 +41,40 @@ module.exports = class ServerRouter {
...
+        const route = await this.#getRouteInfo(reqUrl);
+
+        return _.reduce(route.slots, (routeAssets, slotData) => {
+            const appInfo = this.__apps[slotData.appName];
+
+            if (!routeAssets.includes(appInfo.spaBundle)) {
+                routeAssets.push(appInfo.spaBundle);
+            }
+
+            Object.keys(appInfo.dependencies).filter((dependencyName) => {
```
Comment (StyleT) [inline ilc/common/router/ServerRouter.js]: Why are you using filter method here?

Comment (StyleT) [reply ilc/common/router/ServerRouter.js]: I think you're doing some awkward things here.... Let's discuss...
Simple `Object.assign` should work fine here.

Comment (oleh-momot) [reply ilc/common/router/ServerRouter.js]: Updated

File: ilc/common/router/ServerRouter.js (hunk 1 above)
Comment (oleh-momot) [reply ilc/common/router/ServerRouter.js]: Moved

File: ilc/common/router/ServerRouter.js (hunk 2 above)
Comment (oleh-momot) [reply ilc/common/router/ServerRouter.js]: I removed this check

**Pull request merged**